File: codeproblems/src/main/java/ivanmarkovic/recursion/TowersOfHanoi.java
package java.ivanmarkovic.recursion;

public class TowersOfHanoi {

    public static void towers(int number, String start, String spare, String end){
        if(number == 1)
            printMove(start, end);
        else {
            towers(number - 1, start, end, spare);
            towers(1, start, spare, end);
            towers(number - 1, spare, start, end);
        }
    }

    public static void printMove(String start, String end){
        System.out.println("Moving from " + start + " to " + end);
    }
}

File: recursion/towers-of-hanoi.py

def printMove(start: str, end: str):
    print("Move from", start, "to", end)

def towers(num: int, start: str, spare: str, end: str):
    if num == 1:
        printMove(start, end)
    else:
        towers(num - 1, start, end, spare)
        towers(1, start, spare, end)
        towers(num - 1, spare, start, end)


towers(3, "start", "spare", "end")